src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterToolMonitor.scala

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import akka.actor.{Actor, Props, Timers}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.google.GoogleProjectDAO
+import org.broadinstitute.dsde.workbench.leonardo.config.ClusterServiceConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.ToolDAO
+import org.broadinstitute.dsde.workbench.leonardo.dao.google.GoogleDataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterTool}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterToolMonitor.{DetectClusterStatus, TimerKey, ToolStatus}
+import org.broadinstitute.dsde.workbench.newrelic.NewRelicMetrics
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.model.ClusterTool.Welder
+
+import scala.concurrent.Future
+
+object ClusterToolMonitor {
+
+  def props(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, toolDAOs: Map[ClusterTool, ToolDAO], newRelic: NewRelicMetrics): Props = {
+    Props(new ClusterToolMonitor(config, gdDAO, googleProjectDAO, dbRef, toolDAOs, newRelic))
+  }
+
+  sealed trait ClusterServiceMonitorMessage
+  case object DetectClusterStatus extends ClusterServiceMonitorMessage
+  case object TimerKey extends ClusterServiceMonitorMessage
+
+  case class ToolStatus(val isUp: Boolean, val tool: ClusterTool)
+}
+
+/**
+  * This monitor periodically sweeps the Leo database and checks for clusters which no longer exist in Google.
+  */
+class ClusterToolMonitor(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, toolDAOs: Map[ClusterTool, ToolDAO], newRelic: NewRelicMetrics) extends Actor with Timers with LazyLogging {
+
+  import context._
+
+  override def preStart(): Unit = {
+    super.preStart()
+    timers.startPeriodicTimer(TimerKey, DetectClusterStatus, config.pollPeriod)
+  }
+
+  override def receive: Receive = {
+    case DetectClusterStatus =>
+      for {
+        activeClusters <- getActiveClustersFromDatabase
+        statuses <- Future.traverse(activeClusters)(checkClusterStatus)
+        _ <- Future.traverse(statuses)(handleClusterStatus)
+      } yield ()
+  }
+
+  private def handleClusterStatus(statuses: List[ToolStatus]): Future[Unit] = {
+    statuses.foreach { status =>
+      if (!status.isUp) {
+        val toolName = status.tool.toString
+        logger.info(s"The tool ${toolName} is down")
+        newRelic.incrementCounterIO(toolName + "Down")
+      }
+    }
+    Future.unit
```

Comment:
this will not do anything. You're just constructing a list of `IO`s and they'll never run

Reply:
Ah I was experimenting with incrementCounterFuture but forgot to add the runSync when I changed it back, good catch

Reply:
why not just use https://github.com/DataBiosphere/leonardo/pull/1016#discussion_r312968160 ?